### PR TITLE
Swagger.json OpenAPI file for the allocation/compute API

### LINKED ISSUE #1579: Provide an OpenAPI definition
A published API would enable automatic documentation and SDK generation and facilitate easier testing and integrations. OpenAPI is the preferred standard.

https://oai.github.io/Documentation/specification.html

### PR DESCRIPTION
**Pull request by mattray**
## What does this PR change?
This provides an initial documented swagger.json for use as an OpenAPI spec.

## Does this PR relate to any other PRs?
https://github.com/opencost/opencost-website/pull/70 will need to be closed and replaced with generated documentation.

## How will this PR impact users?
OpenAPI spec may be used for future integrations.

## Does this PR address any GitHub or Zendesk issues?
Fixes https://github.com/opencost/opencost/issues/1579

## How was this PR tested?
Developed within SwaggerHub using Swagger-Inspector, imported into Postman

## Does this PR require changes to documentation?
This will become the basis for generated API documentation.

## Have you labeled this PR and its corresponding Issue as "next release" if it should be part of the next Opencost release? If not, why not?
No. Documentation,

### CODE REVIEW

**Comment by mattray** [comment]
@dwbrown2 @bstuder99 @AjayTripathy this is the initial swagger.json for documenting the `/allocation/compute` API. 

We weren't sure there were any other API endpoints defined in [router.go](https://github.com/opencost/opencost/blob/develop/pkg/costmodel/router.go#L1744) that should be the public API for OpenCost. It was my impression that `/allocation/compute/summary` isn't used by OpenCost, is that correct? The other endpoints appear to be helper/private APIs, I didn't think they needed documenting.

@bstuder99 and I will investigate how to best generate Markdown-based documentation for inclusion on https://opencost.io

**Comment by brstuder** [comment]
Great next step, thanks for doing this

**Comment by dwbrown2** [comment]
Awesome! @teevans I know you have some familiarity here. Would you or anybody on your team be able to provide an initial review?

File: docs/swagger.json [hunk 1]
```diff
@@ -0,0 +1,630 @@
...
+    {
+    "url": "http://localhost:9003",
+    "description": "kubectl port-forward --namespace opencost service/opencost 9003"
+  }
+  ],
+  "paths": {
+    "/allocation/compute": {
+      "get": {
+        "summary": "query for costs and resources allocated to Kubernetes workloads",
+        "description": "The standard OpenCost API query for costs and resources allocated to Kubernetes workloads. You may specify the `window` date range, the Kubernetes primitive `aggregate` on, ",
```
Comment (teevans) [inline docs/swagger.json]: I could be wrong, but it looks like this sentence may have gotten cut off

**Review by teevans — COMMENTED** [review]
Man, this a killer first start. Thank you for the work on this!! I left one comment on a description for a call that may have gotten truncated. Other than that, I think this looks good. for those interested what this would look like in a swagger client, go [Here](https://petstore.swagger.io/) - And replace the URL in the very top text field with the below URL, and click explore. That will bring up the  file here to play with. 

```
https://raw.githubusercontent.com/opencost/opencost/27cc6e36e83d43550d5679cb19d97d0258e1cf31/docs/swagger.json
```

File: docs/swagger.json (hunk 1 above)
Comment (mattray) [reply docs/swagger.json]: Updated in [1bc3043](https://github.com/opencost/opencost/pull/1619/commits/1bc304385b1d5682d618c037678ac92596c96fea)

**Review by teevans — APPROVED** [review]
LGTM

**Pull request merged**